internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -210,6 +212,39 @@ func (c *CodePipeline) ListPipelineNamesByTags(tags map[string]string) ([]string
 	return pipelineNames, nil
 }
 
+// ListPipelineExecution returns the ExecutionID of the most recent execution of a pipeline.
+func (c *CodePipeline) ListPipelineExecution(pipelineName string) (*string, error) {
+	input := &cp.ListPipelineExecutionsInput{
+		MaxResults:   aws.Int64(1),
+		PipelineName: &pipelineName,
+	}
+	output, err := c.client.ListPipelineExecutions(input)
+	if err != nil {
+		return nil, fmt.Errorf("list pipeline execution for %s: %w", pipelineName, err)
+	}
+	return output.PipelineExecutionSummaries[0].PipelineExecutionId, nil
```

Comment:
Will this ever return a zero length array?

Reply:
I added a zero-length check. 
Discussed name-check w/ @iamhopaul123 offline and because the pipeline name isn't part of ListPipelineExecutionsOutput, it's not worth another API call to GetPipelineExecution to confirm.